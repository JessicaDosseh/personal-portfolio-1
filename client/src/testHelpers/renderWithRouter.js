import React from "react";
import { Router } from "react-router-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
