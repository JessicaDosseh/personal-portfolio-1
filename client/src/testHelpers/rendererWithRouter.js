import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
