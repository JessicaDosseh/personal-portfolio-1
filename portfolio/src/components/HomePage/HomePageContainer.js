import React, { Component } from "react";
import { HeaderContainer } from "../HeaderContent";

export default class HomePageContainer extends Component {
  render() {
    return (
      <div>
        <HeaderContainer />
      </div>
    );
  }
}
