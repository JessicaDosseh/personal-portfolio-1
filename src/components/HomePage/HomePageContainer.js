import React from "react";
import { HeaderContainer } from "../HeaderContent";
import { FeaturedWorkContainer } from "../FeaturedWork";
import { Footer } from "../Footer";

const HomePageContainer = props => {
  return (
    <div>
      <HeaderContainer darkmode={props.darkmode} />
      <FeaturedWorkContainer darkmode={props.darkmode} />
      <Footer darkmode={props.darkmode} />
    </div>
  );
};

export default HomePageContainer;
