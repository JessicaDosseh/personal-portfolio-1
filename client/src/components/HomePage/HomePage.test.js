import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import HomePageContainer from "./HomePageContainer";

describe("<HomePageContainer />", () => {
  it("should render the HomePageContainer component without crashing", () => {
    const push = jest.fn();
    const scrollToSpy = jest.fn();
    window.scrollTo = scrollToSpy;
    const helpers = renderWithRouter(
      <HomePageContainer darkmode={false} history={push} />
    );
  });

  it("should match the snapshot of HomePageContainer", () => {
    const tree = rendererWithRouter(<HomePageContainer darkmode={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should display the HeaderContainer component", () => {
    const { getByText } = renderWithRouter(
      <HomePageContainer darkmode={false} />
    );
    expect(getByText(/I'm Nathan Thomas/i)).toBeDefined();
  });

  it("should display the FeaturedWorkContainer component", () => {
    const { getByText } = renderWithRouter(
      <HomePageContainer darkmode={false} />
    );
    expect(getByText(/Featured Projects/i)).toBeDefined();
  });

  it("should display the Footer component", () => {
    const { getByText } = renderWithRouter(
      <HomePageContainer darkmode={false} />
    );
    expect(getByText(/Looking to start a project\?/i)).toBeDefined();
  });
});
