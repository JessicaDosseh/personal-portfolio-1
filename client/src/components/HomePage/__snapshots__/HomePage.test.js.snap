// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePageContainer /> should match the snapshot of HomePageContainer 1`] = `
<div>
  <div
    className="sc-AykKC gvFiiQ"
  >
    <p>
      I'm Nathan Thomas, a
       
      <a
        href="https://github.com/nwthomas"
      >
        full stack software engineer
      </a>
       
      living in the Napa Valley and currently attending
       
      <a
        href="https://lambdaschool.com/"
      >
        Lambda School
      </a>
      . Let's
       
      <a
        href="/contact"
        onClick={[Function]}
      >
        talk
      </a>
      .
    </p>
    <div
      className="sc-AykKD bZWOgB"
    >
      <div>
        <h4>
          Current Student:
        </h4>
        <div>
          <a
            href="https://lambdaschool.com/"
          >
            <img
              alt="Current school logo"
              data-testid="schoolLogo"
              src="lambda-logo-white.png"
            />
          </a>
        </div>
      </div>
      <div>
        <h4>
          Previously:
        </h4>
        <div>
          <a
            href="https://www.adventisthealth.org/"
          >
            <img
              alt="Former employer logo"
              data-testid="workLogo"
              src="ah-logo-white.png"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AykKE iHfZmp"
  >
    <p
      id="featured-work"
    >
      Featured Projects
    </p>
    <div>
      <div>
        <div>
          <a
            href="https://github.com/dustinmyers/react-conflux"
          >
            <img
              alt="Conflux state management sample"
              data-testid="confluxPicture"
              src="conflux-logo.png"
            />
          </a>
          <h4>
            Conflux State Management
          </h4>
        </div>
      </div>
      <div>
        <div>
          <a
            href="https://github.com/weight-lifting-journal-bw-team/back-end-nathan-thomas"
          >
            <img
              alt="weight lifting journal sample"
              data-testid="weightLiftingJournalPicture"
              src="weight-lifting-journal-app.png"
            />
          </a>
          <h4>
            Weight Lifting Journal
          </h4>
        </div>
      </div>
      <div>
        <div>
          <a
            href="https://github.com/nwthomas/tippr"
          >
            <img
              alt="tippr portfolio sample"
              data-testid="tipprPicture"
              src="tippr-macbook.png"
            />
          </a>
          <h4>
            tippr payment app
          </h4>
        </div>
      </div>
      <div>
        <div>
          <a
            href="https://github.com/nwthomas/instagram-clone"
          >
            <img
              alt="Instagram portfolio sample"
              data-testid="instagramPicture"
              src="instagram-macbook.png"
            />
          </a>
          <h4>
            Instagram Clone
          </h4>
        </div>
      </div>
      <div>
        <div>
          <a
            href="https://github.com/nwthomas/huddle-landing-page"
          >
            <img
              alt="Huddle portfolio sample"
              data-testid="huddlePicture"
              src="huddle-landing-page.png"
            />
          </a>
          <h4>
            Huddle Landing Page
          </h4>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AykKF joCMMf"
  >
    <div>
      <p>
        Looking to start a project?
      </p>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Get in contact.
      </a>
    </div>
    <div
      className="sc-AykKG bjsjyh"
    >
      <a
        href="https://github.com/nwthomas"
      >
        GitHub
      </a>
      <a
        href="https://www.linkedin.com/in/nwthomas-profile/"
      >
        LinkedIn
      </a>
      <a
        href="https://medium.com/@nwthomas"
      >
        Medium
      </a>
      <a
        href="https://twitter.com/nwthomas_"
      >
        Twitter
      </a>
      <a
        href="https://www.instagram.com/nwthomas/"
      >
        Instagram
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Email
      </a>
    </div>
  </div>
  <div
    className="sc-AykKH cDjEel"
  >
    <h4>
      Copyright Â© 2019 Nathan Thomas
    </h4>
  </div>
</div>
`;
